modules/apps/headless/headless-admin-site/headless-admin-site-impl/src/main/java/com/liferay/headless/admin/site/internal/resource/v1_0/NavigationMenuResourceImpl.java

```diff
@@ -584,6 +604,188 @@ private int _getNavigationMenuType(NavigationMenu navigationMenu) {
 		return siteNavigationMenuItemsMap;
 	}
 
+	private String _getTypeSettings(NavigationMenuItem navigationMenuItem) {
+		UnicodeProperties unicodeProperties = UnicodePropertiesBuilder.create(
+			true
+		).put(
+			"defaultLanguageId", navigationMenuItem.getDefaultLanguageId()
+		).put(
+			"displayIcon", navigationMenuItem.getDisplayIcon()
+		).put(
+			"useCustomName",
+			() -> {
+				if (navigationMenuItem.getUseCustomName() != null) {
+					return String.valueOf(
+						navigationMenuItem.getUseCustomName());
+				}
+
+				return null;
+			}
+		).build();
+
+		Object navigationMenuItemSettings =
+			navigationMenuItem.getNavigationMenuItemSettings();
+
+		if (navigationMenuItemSettings != null) {
+			String localizedNames = _getLocalizedNamesFromI18nMap(
+				navigationMenuItem.getName_i18n());
+
+			Map<String, String> localizedPropertyMap =
+				_getLocalizedPropertiesMapFromI18nMap(
+					navigationMenuItem.getName_i18n());
+
+			String type = navigationMenuItem.getType();
+
+			if (Objects.equals(
+					type,
+					SiteNavigationMenuItemTypeConstants.ASSET_VOCABULARY)) {
```

Comment:
Like the previous commit, is there a way to have the logic encapsulated in its respective class or is this the pattern used elsewhere?